File: Ex3Num8.kt
fun main() {
    var sum = 0

    print("Введите количество чисел: ")
    val n = readln().toInt()

    for (i in 1..n) {
        print("Введите число $i: ")
        var number = readln().toInt()

        // Обрабатываем отрицательные числа
        if (number < 0) {
            number = -number
        }

        // Разбираем число по цифрам
        var temp = number
        while (temp > 0) {
            val digit = temp % 10
            if (digit % 3 == 0 && digit != 0) {
                sum += digit
            }
            temp /= 10
        }
    }

    println("Сумма цифр, делящихся на 3: $sum")
}

File: Ex3Num8.rb
sum = 0

print "Введите количество чисел: "
n = gets.chomp.to_i

(1..n).each do |i|
  print "Введите число #{i}: "
  number = gets.chomp.to_i.abs  # Модуль числа для обработки отрицательных

  # Разбираем число по цифрам
  temp = number
  while temp > 0
    digit = temp % 10
    sum += digit if digit % 3 == 0 && digit != 0
    temp /= 10
  end
end

puts "Сумма цифр, делящихся на 3: #{sum}"